autoload/infowindow.vim

```diff
@@ -1,34 +1,64 @@
-let g:_infobuf_1 = -1
+let g:infowindow_buffnr = -1
+let g:infowindow_loaded = 1
 
-function infowindow#toggle()
-  if (bufexists(g:_infobuf_1))
-    execute 'bdelete ' g:_infobuf_1
-    let g:_infobuf_1 = -1
-    return
+function infowindow#destroy()
+  if (bufexists(g:infowindow_buffnr))
+    execute 'bdelete ' . g:infowindow_buffnr
+    let g:infowindow_buffnr = -1
+    unlet g:infowindow_mainwindow
   endif
-  let cbuf = bufnr('')
-  let buffilename = bufname(cbuf)
-  let filename = ' name:   ' . (strlen(buffilename) > 0 ? buffilename : '[No Name]') . ' '
-  let filetype = ' type:   ' . (strlen(&filetype) > 0 ? &filetype : 'unknown') . ' '
-  let fileformat = ' format: ' . &fileformat . ' '
-  let numlines = ' lines:  ' . line('$') . ' '
+endfunction
+
+function infowindow#timer_handler(timer)
```

Comment:
Some functions are probably not needed to be public if they are meant to be used internally. So instead of `infowindow#timer_handler` you can name it `s:time_handler` and call it like `<SID>time_handler`

Reply:
I could not convert `timer_handler` to a local function. `timer_start` does not allow it.